assets/src/helpers.js

```diff
@@ -240,6 +244,27 @@ export const addAMPExtraProps = ( props, blockType, attributes ) => {
 		...ampAttributes,
 	};
 };
+
+/**
+ * Wraps all movable blocks in a grid layer.
+ *
+ * @param {Object} element
+ * @param {Object} blockType
+ *
+ * @return {Object} The element.
+ */
+export const wrapBlocksInGridLayer = ( element, blockType ) => {
+	if ( ! ALLOWED_MOVABLE_BLOCKS.includes( blockType.name ) ) {
+		return element;
+	}
+
+	return (
+		<amp-story-grid-layer template="vertical">
```

Comment:
Is it expected that every block is wrapped in an `amp-story-grid-layer` when looking at the source of an `amp-story`?

```html
<!-- wp:amp/amp-story-page {"autoAdvanceAfterDuration":0} -->
<amp-story-page style="background-color:#ffffff" class="wp-block-amp-amp-story-page">
    <amp-story-grid-layer template="fill"></amp-story-grid-layer>
    
    <!-- wp:amp/amp-story-text {"placeholder":"Write something!","tagName":"h1","autoFontSize":28,"positionTop":10} -->
    <amp-story-grid-layer template="vertical">
        <h1 style="font-size:28px;width:76%;height:9%;position:absolute;top:10%;left:5%" class=""
            id="46f4eecd-32c5-4ba0-8ec4-9e6c6b69cffb">
            <amp-fit-text layout="fill" class="amp-text-content">asdasdsad</amp-fit-text>
        </h1>
    </amp-story-grid-layer>
    <!-- /wp:amp/amp-story-text -->

    <!-- wp:amp/amp-story-text {"tagName":"h1","autoFontSize":28,"positionTop":20} -->
    <amp-story-grid-layer template="vertical">
        <h1 style="font-size:28px;width:76%;height:9%;position:absolute;top:20%;left:5%" class=""
            id="dc2f83b8-d6d7-4989-9299-df8c6914dfa1">
            <amp-fit-text layout="fill" class="amp-text-content">asdasdasd</amp-fit-text>
        </h1>
    </amp-story-grid-layer>
    <!-- /wp:amp/amp-story-text -->

    <!-- wp:image {"id":151,"positionTop":31,"positionLeft":12} -->
    <amp-story-grid-layer template="vertical">
        <figure class="wp-block-image" id="0da70857-1da6-4b19-b1be-bce2a7b4ac80"
                style="position:absolute;top:31%;left:12%">
            <img src="http://wordpressdev.lndo.site/content/uploads/2019/02/PIA19048_realistic_color_Europa_mosaic_alt-1024x757.jpg"
                 alt="" class="wp-image-151"/>
            <figcaption>converted PNM file</figcaption>
        </figure>
    </amp-story-grid-layer>
    <!-- /wp:image --> 
</amp-story-page>
<!-- /wp:amp/amp-story-page -->
```

Reply:
See 2d93c407483aa1d753d62c4668fa81264cc8b6b2 for this workaround.

Reply:
> Note the line break at the beginning.

Looking at the `post_content` you shared, that line break is a bit unexpected.

Regarding the `amp-story-grid-layer`, it is what I came up with in order to easily support the CTA layer within `InnerBlocks`. If you have another idea there, please let me know.